tests/test_dithering_binning.py

```diff
@@ -0,0 +1,169 @@
+
+    def test_dithering_balanced_values(self):
+        """Normal values to test"""
+        self.db_object.binning(self.x, self.weights, self.labels, self.label_length)
+        self.assertEqual(0, self.db_object.bins[0].coins[0].value)
+        self.assertEqual(1, self.db_object.bins[0].coins[1].value)
+        self.assertEqual(2, self.db_object.bins[0].coins[2].value)
+        self.assertEqual(3, self.db_object.bins[1].coins[3].value)
+        self.assertEqual(4, self.db_object.bins[1].coins[4].value)
+        self.assertEqual(5, self.db_object.bins[1].coins[5].value)
+        self.assertEqual(6, self.db_object.bins[2].coins[6].value)
+        self.assertEqual(7, self.db_object.bins[2].coins[7].value)
+        self.assertEqual(8, self.db_object.bins[2].coins[8].value)
+
+    def test_labeling(self):
+        """Check if Labels came out right"""
+        self.x.append(float('nan'))  # Handling None and Nan Values
+        self.x.append(None)
+        self.weights.append(1)
+        self.weights.append(1)
+        self.db_object.binning(self.x, self.weights, self.labels, self.label_length)
+
+        self.assertEqual('b1', self.db_object.label[0])
+        self.assertEqual('b1', self.db_object.label[1])
+        self.assertEqual('b1', self.db_object.label[2])
+        self.assertEqual('b1', self.db_object.label[3])
+        self.assertEqual('b1', self.db_object.label[4])
+        self.assertEqual('b2', self.db_object.label[5])
+        self.assertEqual('b2', self.db_object.label[6])
+        self.assertEqual('b3', self.db_object.label[7])
+        self.assertEqual('b3', self.db_object.label[8])
+        self.assertEqual('NaN', self.db_object.label[9])
+        self.assertEqual('NaN', self.db_object.label[10])
+
+    def test_db_value_inbalance(self):
+        """"Inbalance of values"""
+        self.x.append(1)
+        for i in range(0, 45):
+            self.x.append(9)
+        for i in range(0, 45):
+            self.x.append(10)
+        self.weights = [1] * len(self.x)
+        self.db_object.binning(self.x, self.weights, self.labels, self.label_length)
+
+        for k, coin in self.db_object.bins[0].coins.items():
+            if coin.value == 10:  # No values should be 10
+                self.assertTrue(False)
+        for k, coin in self.db_object.bins[1].coins.items():
+            if coin.value < 9:  # Values should only be 9 or 10
+                self.assertTrue(False)
+        for k, coin in self.db_object.bins[2].coins.items():
+            if coin.value != 10:  # Values can only be 10
+                self.assertTrue(False)
+
+    def test_db_weight_inbalance(self):
+        """Inbalance of weight on one side"""
+        for i in range(0, 3):
+            self.x.append(i+8)
+            self.weights.append(3)
+        self.db_object.binning(self.x, self.weights, self.labels, self.label_length)
+
+        for i in range(0, self.label_length):
+            self.assertTrue(self.db_object.bins[i].weight == 6)
```

Comment:
You almost always want to use `assertEqual` over `assertTrue` with a few exceptions